bluebottle/funding/views.py

```diff
@@ -177,19 +177,23 @@ class PayoutDetails(JsonApiViewMixin, AutoPrefetchMixin, RetrieveUpdateAPIView):
     permission_classes = (IsFinancialMember,)
 
     def perform_update(self, serializer):
-        status = serializer.validated_data['status']
-        # related to this Funding.
-        payout = serializer.instance
-        if status == 'started':
-            payout.states.start(save=True)
-        if status == 'scheduled':
-            payout.states.start(save=True)
+        status = serializer.validated_data.pop('status')
+        if status == 'reset':
+            serializer.instance.states.reset()
         if status == 'new':
-            payout.states.reset(save=True)
+            serializer.instance.states.schedule()
+        if status == 'scheduled':
+            serializer.instance.states.schedule()
+        if status == 're_scheduled':
+            serializer.instance.states.schedule()
+        if status == 'started':
+            serializer.instance.states.start()
         if status == 'succeeded':
-            payout.states.succeed(save=True)
-        if status == 'confirm':
-            payout.states.succeed(save=True)
+            serializer.instance.states.succeed()
+        if status == 'confirmed':
+            serializer.instance.states.succeed()
+        print status
```

Comment:
This should not be there.

Maybe group the same call in the same if with `if status in ('new', 'scheduled', 're_scheduled', ):`

Reply:
It's informative!